consensus/db/db.go

```diff
@@ -60,11 +60,26 @@ func (w *walEntry[V, H, A]) UnmarshalCBOR(data []byte) error {
 }
 
 // TendermintDB defines the methods for interacting with the Tendermint WAL database.
+//
+// The purpose of the WAL is to record any event that may result in a state change.
+// These events fall into the following categories:
+// 1. Incoming messages. We do not need to store outgoing messages.
+// 2. When we propose a value.
+// 3. When a timeout is triggered (not when it is scheduled).
+// The purpose of the WAL is to allow the node to recover the state it was in before the crash.
+// No new messages should be broadcast during replay.
+//
+// We commit the WAL to disk when:
+// 1. We start a new round
+// 2. Right before we propose a block
+// 3. When consensus is reached
+//
+// We call Delete when we start a new height and commit a block
 type TendermintDB[V types.Hashable[H], H types.Hash, A types.Addr] interface {
-	// CommitBatch writes the accumulated batch operations to the underlying database.
-	CommitBatch() error
+	// FlushWAL writes the accumulated batch operations to the underlying database.
+	FlushWAL() error
 	// GetWALMsgs retrieves all WAL messages (consensus messages and timeouts) stored for a given height from the database.
-	GetWALMsgs(height types.Height) ([]walEntry[V, H, A], error)
+	GetWALMsgs(height types.Height) ([]WalEntry[V, H, A], error)
```

Comment:
I'm wondering if we need to access `WalEntry` from other packages. What do you think if we keep it private and extract the message from `walEntry`?
```suggestion
	GetWALMsgs(height types.Height) ([]Message[V, H, A], error)
```

Reply:
Nevermind, I forgot the timeout case.